### PR TITLE
Update table styling and spacing

### PR DESCRIPTION
**Pull request by janfaracik**
old
new

This is a small PR just to add some more polish to the new table introduced in #5851.

* Minor padding changes to the table 
* The corners are now rounded correctly (there was a mismatch before)
* Links/buttons now have the same focus state as components in #5923 
* Build status now uses the updated colours introduced in #5851 
* Minor change to the users pages - descriptions are now styled correctly

### Proposed changelog entries

* Update table styling and spacing with small table UI tweaks.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/resources/lib/hudson/projectViewRow.jelly [hunk 1]
```diff
@@ -34,8 +34,5 @@ THE SOFTWARE.
     <j:forEach var="col" items="${columnExtensions}">
       <st:include page="column.jelly" it="${col}"/>
     </j:forEach>
-    <td>
```
Comment (janfaracik) [inline core/src/main/resources/lib/hudson/projectViewRow.jelly:37]: There was an extra row/column for the dashboard table causing unnecessary padding/spacing. Removing it doesn't seem to have broken anything - anybody know what it was there for?

Comment (NotMyFault) [reply core/src/main/resources/lib/hudson/projectViewRow.jelly:37]: That's basically always been like that; the last part was a sortable column. Though it doesn't sort anything, likely used for spacing in past days, I guess?

Comment (daniel-beck) [reply core/src/main/resources/lib/hudson/projectViewRow.jelly:37]: Looks like initially (pre-dating configurable columns) it was a spacer between the build icon and the data columns: https://github.com/jenkinsci/jenkins/blob/f7d7507f39de655c11fc28501ef3732304dc9a79/core/src/main/resources/lib/hudson/projectView.jelly#L118-L127

When https://github.com/jenkinsci/jenkins/commit/6f5dafd34320a89f1dfd1c48081eff4f6ee5a228 made the default columns part of the extension system, it was left in.

**Review by NotMyFault — APPROVED** [review]
Looks good, though I had to take a closer look to notice the even more rounder table frames :P

**Review by timja — APPROVED** [review]

**Pull request merged**